trieste/models/interfaces.py

```diff
@@ -99,6 +99,14 @@ def get_kernel(self) -> gpflow.kernels.Kernel:
         """
         raise NotImplementedError(f"Model {self!r} does not provide a kernel")
 
+    def log(self, context: str) -> None:
```

Comment:
That's really minor, and probably down to taste, but for me "log" normally means text-based logging, and this feel more like reporting metrics.

Reply:
What would you suggest calling it? At the moment it's as genera as users want it to be, though I expect after we play with this for a while we may end up refactoring out some metric specific logic like hrvoje suggested.

Reply:
I am not sure if it is useful to abstract to any kind of logging here. E.g. with usual logging it isn't uncommon to do thing like
```
logger.log(f"Calling optimizer for {self.model} and dataset {dataset}", DEBUG)
```
so some very low level stuff that one wouldn't want to send to tensorboard. However we are not using Python logging (yet?), so all that is just hypothetical. Whereas this `model.log` has very specific purpose of writing numerical data to graph. Unless I misunderstood it, which is entirely possible of course.

I guess my suggestion here is to be a bit more specific with this method. `(log|write)_to_tensorboard`?

Reply:
actually, files that are saved don't necessarily need to be used by tensorboard, you could extract the data and make your own plots etc, so I think it can stay more general, but I agree with Andrei that there might be a confusion with usual python logging which is not the intended purpose - I would vote for metrics

Reply:
model.log doesn't have to be used just to write metrics to tensorboard: you could use it for any sort of model logging (in the broad sense) that you want to happen after every BO loop. Though as I said, I expect we'll refactor the code soon to handling metrics more explicit, at which point this will be renamed.

Reply:
ok, let's leave it for now

Reply:
Happy to rename it though. Though it should be a verb, as it's writing metrics, not returning them. `write_metrics`?

Reply:
`generate_metrics`?

Reply:
write_metrics is more precise